File: src/gem5_hw/basic_harvest.cc
//
// @author: Duan Xg
// 
//

#include "basic_harvest.hh"
#include "debug/EnergyMgmt.hh"

double BasicHarvest::energy_harvest(double energy_harvested, double energy_remained)
{
	if(energy_remained + energy_harvested > _capacity)
	{
		DPRINTF(EnergyMgmt,"energy full: %f+%f->%f\n",energy_remained,energy_harvested,_capacity);
		return _capacity;
	}
    return (energy_remained + energy_harvested);
}

void BasicHarvest::init()
{
    DPRINTF(EnergyMgmt, "basic energy harvest module is used in this system.\n");
}


BasicHarvest *
BasicHarvestParams::create()
{
    return new BasicHarvest(this);
}

File: src/gem5_hw/BasicHarvest.py
from m5.SimObject import SimObject
from BaseHarvest import BaseHarvest
from m5.params import *
from m5.proxy import *

class BasicHarvest(BaseHarvest):
    type = 'BasicHarvest'
    cxx_header = "gem5_hw/basic_harvest.hh"
    capacity = Param.Float(50000, "capacity of energy system, default 50000")